doc/lsst.ap.pipe/pipeline-with-fakes.rst

```diff
@@ -0,0 +1,194 @@
+
+The ``SConstruct`` file defines the build process for generating the ApPipeWithFakes pipeline, this build process:
+
+1. **Takes the base ApPipe pipeline** (``ApPipe.yaml``) as input
+2. **Uses the ``make_injection_pipeline`` command** to automatically generate the fakes version
+3. **Adds injection-specific tasks** from ``ApPipePostInjectedTasks.yaml``
+4. **Applies systematic transformations**:
+
+   * Prefixes task names with ``fakes_`` to create parallel processing paths
+   * Configures injection parameters (PSF, photometry, WCS settings)
+   * Sets up proper data connections between injected and standard processing
+   * Creates analysis tasks for fake source performance evaluation
+
+5. **Ensures proper build dependencies**: The generated file depends on the version target and is required for tests
+
+Key Build Parameters
+~~~~~~~~~~~~~~~~~~~~
+
+* **Target exposure**: ``preliminary_visit_image`` - the image type to inject sources into
+* **Subsets processed**: ``apPipe`` and ``prompt`` - which pipeline subsets get fakes versions
+* **Configuration overrides**: Sets injection task defaults for PSF, photometry, and WCS handling
+* **APDB configuration**: Uses placeholder config that must be user-specified at runtime
+
+Why Generate at Build Time?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This approach provides several advantages:
+
+* **Consistency**: Ensures the fakes pipeline stays synchronized with the base pipeline
+* **Maintenance**: Automatically incorporates changes to the base pipeline without manual updates
+* **Flexibility**: Allows systematic application of injection patterns across different pipeline subsets
+* **Testing**: Ensures the fakes pipeline is always available for validation and testing
+
+The generated ``ApPipeWithFakes.yaml`` file should **not** be manually edited, as it will be
+overwritten during the next package build. Instead, modifications should be made to:
+
+* The base ``ApPipe.yaml`` pipeline
+* The ``ApPipePostInjectedTasks.yaml`` injection-specific tasks
+* The ``SConstruct`` build configuration
+* Camera-specific configuration files that import the ingredients
+
+Running the Pipeline
+====================
+
+Prerequisites
+-------------
+Running `ApPipeWithFakes` requires the same prerequisites as the standard `ApPipe` pipeline, and in addition an **injection catalog**: pre-made injection catalogs should be ingested into the input collection to use.
+
+Output Products
+===============
+
+The ApPipeWithFakes pipeline produces all standard AP pipeline outputs plus additional fake-specific datasets:
+
+Standard AP Products (with ``fakes_`` prefix)
+----------------------------------------------
+
+* ``fakes_preliminary_visit_image``: Calibrated science image with injected sources
+* ``fakes_difference_image``: Difference image between injected science image and template
+* ``fakes_dia_source_detector``: Detected sources in difference image
+* ``fakes_dia_object_apdb``: Associated DiaObjects stored in APDB
```

Comment:
I'm honestly not sure it's worth documenting this, especially not in multiple places. The AP pipeline produces dozens of output datasets, and the list is growing. Better to rely on automated tools.